torch/jit/__init__.py

```diff
@@ -325,19 +335,22 @@ def verify(model, args, loss_fn=torch.sum, devices=None):
 
     # TODO: Consider adding a utility function to torch.jit to test
     # for this case
-    if not hasattr(model, 'compiled_fn') or not isinstance(model.compiled_fn, torch._C.CompiledFunction):
+    if not isinstance(model, torch._C.CompiledFunction):
         raise TypeError("Cannot verify an uncompiled module.  Add @torch.jit.compile to compile it")
+    is_module = isinstance(model, Module)
 
     if not isinstance(args, tuple):
         args = (args,)
 
     saved_args = _clone_inputs(args)
-    saved_state = copy.deepcopy(model.state_dict())
+    if is_module:
```

Comment:
how come we gained these if statements? I thought we decided that functions would get turned into parameter-less modules when compiled to avoid polymorphism like this.

Reply:
I changed it, so that they no longer turn into modules. I think that's just cleaner - we don't allow capturing parameters in compiled functions anyway, and I'd rather have the decorators to not add any special methods to objects that didn't have them. I don't think it will make the code much more complicated. It's just those few `if`s in `verify`.